portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -56,18 +56,17 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
  
         List<String> pastComments = new ArrayList<>();
  
+        int numCommentsWanted = 0;
+        boolean setNumCommentsWanted = true;
         for (Entity entity : results.asIterable()) {
-            int numCommentsWanted = 0;
-            boolean setNumCommentsWanted = true;  
-            
-            // Grab number of comments wanted to be seen by user only in the first entity
-            if (numCommentsWantedTrigger) {
-                numCommentsWanted = validNumInput;
-                setNumCommentsWanted = false;
-            }          
-            
             String comment = (String) (entity.getProperty("newComment"));
             pastComments.add(comment);
+
+            // Grab number of comments wanted to be seen by user only in the first entity
+            if (setNumCommentsWanted) {
+                numCommentsWanted = Integer.parseInt((String)(entity.getProperty("seeNumComments")));
+                setNumCommentsWanted = false;
```

Comment:
I think you can just use a break here to break out of the loop instead of using setNumCommentsWanted.

Reply:
If I used break it would break out of the if statement and the for loop. However, I need to keep looping after I retrieve the first entity's number of comments wanted so that I can add the rest of the entities' comments to the pastComments arraylist. So I'm not sure how to use break if you could explain more?

Second way would be to somehow get only the first entity out of results outside of the for loop which I can't find how to do.

Reply:
Ooh, I see what you're trying to do now. I think the setNumCommentsWanted variable was making me confused.
Maybe desiredNumCommentsFound would be clearer? And have it be false first.

To answer your question (but not to recommend doing this): I think you can use PreparedQuery's asList (https://cloud.google.com/appengine/docs/standard/java/javadoc/com/google/appengine/api/datastore/PreparedQuery) to get the results as a list and then get the first entity using the list, but that may be inefficient since you're already iterating though the elements.

Reply:
I like the variable name change!
Thank you for the resource, I may use asList in the future if I add the feature to delete specific comments.

Reply:
Yup, looking at the documentation is always helpful to figure out what method there are!